src/SwcCompiler.ts

```diff
@@ -103,13 +114,14 @@ export class SwcCompiler implements Compiler {
     return { root, fileNames, config };
   }
 
-  private async buildFile(filename: string, root: string): Promise<CompiledFile> {
+  private async buildFile(filename: string, root: string, config: Config): Promise<CompiledFile> {
     const output = await transformFile(filename, {
       cwd: root,
       filename: filename,
       root: this.workspaceRoot,
       rootMode: "root",
       sourceMaps: "inline",
+      swcrc: false,
```

Comment:
We will not load `.swcrc` files at all. I had to choose between having the config be part of `esbuild-dev.js` or standaline in `.swcrc`. 

This was a hard choice, and I'm not entirely confident in it. One the one hand, using `.swcrc` files would make it obvious that we're supporting the same mechanics as `swc`. On the other, it means yet-another-config-file to track.

I chose this way because `swc` merges configs by giving priority to what's passed in the `transformFile` arguments, meaning that we can't use programmatic defaults and override then in the `.swcrc` file. Should we even have defaults tho? Maybe a few like `sourceMaps`, but `env.target`, or `module.type`?

Reply:
I think that the defaults are indeed a little confusing but I ended up adding defaults to the esbuild compiler so that niceities that make a bit more sense when working in node were on by default like the target and sourcemaps. I can see why esbuild itself has no defaults for those things, but I am ok editorializing a bit since we know more about what platform we're targeting. 

For the swcrc file, I kinda think it might be good to support it such that other uses of `swc` within the same project can take advantage of a shared config. For Gadget, I want us to use `@swc/jest` to transpile our test code, which sadly needs to use a jest specific thing and can't use `esbuild-dev`. That thing ideally will use a `.swcrc` file, so it'd be nice if esbuild-dev was just a client of it as well. I value this the most out of all these aspects I think. 

What do you think about only passing our programmatic defaults if there is no `.swcrc` file, and then documenting what the defaults are in the readme so if folks want to override it they know where to start from?

Reply:
Does that mean we have 3 levels configs tho?
- `esbuild-dev.js`
- `.swcrc`
- programmatic default

I'm a bit hesitant towards that.

If it's specifically for `swc/jest`, it looks like we could reuse the `esbuild-dev.js`'s exported config, as documented in the [Usage](https://github.com/swc-project/jest#usage) section.

Alternatively, we could make such that in `esbuild-dev.js`, `"swc": "import"` would load the `.swcrc` file such that we only have 2 layers of config.

Allowing the `.swcrc` file will probably add a small cost to compilation because we have to traverse the fs from the file to the project root to find an `.swcrc` but we'll only know how much after benchmarking. I'm not expecting anything too dramatic.

Reply:
I'm going to start on that last thing, having `esbuild-dev.js` with the `"swc": "import"` option

Reply:
I am into that. I think most node tools don't pay this much attention and there are always 18 ways to configure them and the precedence is never clear. I like that being explicit :+1:

Reply:
Done